### ISSUE TITLE
Fix up book preferences dialog

### ISSUE THREAD

**Issue by szarroug3** [szarroug3/X-Ray_Calibre_Plugin#45]
Dialog is a bit cramped. Labels are often wider than the boxes/buttons allow - and with the new goodreads URL being so huge this is even worse.

Some options:

```
strip the arguments off the end of the URL found from the search (everything up to and including ?) - this isn't beneficial to us, and is just confusing things/making the URL longer.
```

import urlparse
url = urlparse.urlunparse(urlparse.urlparse(url)._replace(query=None))

```
what is the URL editbox for? If I C&P a URL I've found myself then I don't really need to see the value at all - so only being able to part see it isn't important. If the plugin has found it for me, then I can't tell anything useful from the URL - I'll either suck it and see if it works (in which case, again I don't care about reading the URL) or I'm going to open the URL in my browser and see if it looks sane.
```

So we could either:

```
offer the user a digested version of the page we've found, to check they are happy (ie. yank out title, author, year, etc.)
offer the user a button to click to open the URL in a browser to verify
```

So being able to read the URL isn't important, especially if we add some better way to validate it.

**Comment by stoduk** [comment]
Going for the 80/20 rule on this - I've got a fix to add buttons for ASIN/goodreads URL to open them in a web browser.  It then doesn't matter if we can't see most of the URL because we mostly wouldn't be looking at it.

It might be nice to do the parser idea I mentioned (ie. parse the book, present the info so the user can confirm they like it) but TBH I'd say we have plenty of things that were higher priority right now.

Thoughts?
![screenshot 2016-06-28 13 28 12](https://cloud.githubusercontent.com/assets/5287714/16415235/35cb6cc0-3d34-11e6-96c8-221d8a371175.png)

**Comment by szarroug3** [comment]
Yeah, I think having an open button is sufficient for now. We can always
come back to this and make it better once we've gotten all the higher
priority stuff worked out.

**Comment by szarroug3** [comment]
So this seems to open it specifically for co.uk -- we should probably have it open for whatever region the user is in

**Comment by stoduk** [comment]
Any idea how to do that?  I can only find wordpress plugins to point affiliate links to the local shop, nothing more generic.  Obviously a config preference would be one way, though that would require the user to set that up..

**Comment by szarroug3** [comment]
not sure, the search for asin function also just uses amazon.com rather than location specific.. i'm looking into it

**Comment by szarroug3** [comment]
this is also not that big of a deal.. if there's no simple way to do it, we come back to it

**Comment by stoduk** [comment]
Tracking the extra fix via #76.
